Envconfig fails to parse NetworkServices field of config from env

## Expected Behavior

Cmd-nsc should correctly start and parse config from env.

## Current Behavior

Cmd-nsc fails to parse NetworkServices field of `config.Config` from env.

## Failure logs

```
INFO[0000] Starting NetworkServiceMesh Client ...       
Oct 19 18:44:57.145 [INFO] Creating logger from config: &{nsc@ubuntu false false [] 0xc00032c060 0xc00032c0c0 <nil> <nil> <nil>}
2020/10/19 18:44:57 Initializing logging reporter
This application is configured via the environment. The following environment
variables can be used:

KEY                       TYPE                                            DEFAULT                                           REQUIRED    DESCRIPTION
NSM_NAME                  String                                          nsc                                                           Name of Network Service Client
NSM_CONNECT_TO            URL                                             unix:///var/lib/networkservicemesh/nsm.io.sock                url to connect to NSM
NSM_MAX_TOKEN_LIFETIME    Duration                                        24h                                                           maximum lifetime of tokens
NSM_ROUTES                Comma-separated list of String                                                                                A list of routes asked by client
NSM_LABELS                Comma-separated list of String                                                                                A list of client labels with format key1=val1,key2=val2, will be used a primary list for network services
NSM_MECHANISM             String                                          kernel                                                        Default Mechanism to use, supported values kernel,memif
NSM_NETWORK_SERVICES      Comma-separated list of NetworkServiceConfig                                                                  A list of Network Service Requests with format [{mechanism}]?:${nsName}[@domainName]?/${interfaceName/memIfSocketName}?${label1}=${value1}&${label2}=${value2}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x30 pc=0x932ef8]

goroutine 1 [running]:
github.com/networkservicemesh/cmd-nsc/pkg/config.(*NetworkServiceConfig).UnmarshalBinary(0x0, 0xc00033b4a0, 0x1d, 0x20, 0x1d, 0x20)
        /home/tigrulya/go/src/cmd-nsc/pkg/config/config.go:77 +0x98
github.com/kelseyhightower/envconfig.processField(0xc00003e055, 0x1d, 0xb64c00, 0xc000368090, 0x196, 0xc000368090, 0x196)
        /home/tigrulya/go/pkg/mod/github.com/kelseyhightower/envconfig@v1.4.0/envconfig.go:255 +0x11f4
github.com/kelseyhightower/envconfig.processField(0xc00003e055, 0x1d, 0xb025a0, 0xc0003acce8, 0x197, 0x0, 0x0)
        /home/tigrulya/go/pkg/mod/github.com/kelseyhightower/envconfig@v1.4.0/envconfig.go:312 +0xfa5
github.com/kelseyhightower/envconfig.Process(0xc25013, 0x3, 0xb3dd20, 0xc0003acc00, 0x0, 0x0)
        /home/tigrulya/go/pkg/mod/github.com/kelseyhightower/envconfig@v1.4.0/envconfig.go:215 +0x4b6
main.main()
        /home/tigrulya/go/src/cmd-nsc/main.go:98 +0x48d


```

## Steps to Reproduce

<!--- Please provide detailed steps for reproducing the issue: -->

1. Set `NSM_NETWORK_SERVICES` enviromental variable as `kernel://my-service/nsmKernel`
2. Run cmd-nsc

--- a/main_test.go
+++ b/main_test.go
@@ -21,8 +21,11 @@ package main_test
 import (
 	"context"
 	"net/url"
+	"os"
 	"testing"
 	"time"
+
+	"github.com/kelseyhightower/envconfig"
 
 	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/common"
 	"github.com/networkservicemesh/sdk/pkg/networkservice/common/mechanisms/sendfd"
@@ -72,11 +75,28 @@ func (c *checkConnectionClient) Close(ctx context.Context, conn *networkservice.
 var _ networkservice.NetworkServiceClient = (*nsmTestClient)(nil)
 var _ networkservice.NetworkServiceClient = (*checkConnectionClient)(nil)
 
-func parse(t *testing.T, u string) *config.NetworkServiceConfig {
-	c := &config.NetworkServiceConfig{}
+func parse(t *testing.T, u string) config.NetworkServiceConfig {
+	c := config.NetworkServiceConfig{}
 	err := c.UnmarshalBinary([]byte(u))
 	require.NoError(t, err)
 	return c
+}
+
+func TestParseUrlsFromEnv(t *testing.T) {
+	err := os.Setenv("NSM_NETWORK_SERVICES", "kernel://my-service/nsmKernel,memif://second-service/memif.sock?key=value")
+	require.NoError(t, err)
+
+	c := &config.Config{}
+	err = envconfig.Process("nsm", c)
+	require.NoError(t, err)
+
+	require.Equal(t, 2, len(c.NetworkServices))
+	require.Equal(t, kernel.MECHANISM, c.NetworkServices[0].Mechanism)
+	require.Equal(t, "nsmKernel", c.NetworkServices[0].Path[0])
+
+	require.Equal(t, memif.MECHANISM, c.NetworkServices[1].Mechanism)
+	require.Equal(t, "memif.sock", c.NetworkServices[1].Path[0])
+	require.Equal(t, "value", c.NetworkServices[1].Labels["key"])
 }
 
 func TestParseNSMUrl(t *testing.T) {
@@ -126,7 +146,7 @@ func TestConnectNSM(t *testing.T) {
 			Scheme: "unix",
 			Path:   "/file.sock",
 		},
-		NetworkServices: []*config.NetworkServiceConfig{
+		NetworkServices: []config.NetworkServiceConfig{
 			parse(t, "kernel://my-service/nsmKernel?"),
 		},
 	}
@@ -147,7 +167,7 @@ func TestConnectNSMGRPC(t *testing.T) {
 			Scheme: "unix",
 			Path:   "/file.sock",
 		},
-		NetworkServices: []*config.NetworkServiceConfig{
+		NetworkServices: []config.NetworkServiceConfig{
 			parse(t, "kernel://my-service/nsmKernel?"),
 		},
 	}
@@ -180,7 +200,7 @@ func TestSendFd(t *testing.T) {
 	cfg := &config.Config{
 		Name:      "nsc",
 		ConnectTo: url.URL{Scheme: "tcp", Host: "127.0.0.1:0"},
-		NetworkServices: []*config.NetworkServiceConfig{
+		NetworkServices: []config.NetworkServiceConfig{
 			parse(t, "kernel://my-service/nsmKernel?"),
 		},
 	}

--- a/pkg/config/config.go
+++ b/pkg/config/config.go
@@ -43,7 +43,7 @@ type Config struct {
 	Labels    []string `default:"" desc:"A list of client labels with format key1=val1,key2=val2, will be used a primary list for network services" split_words:"true"`
 	Mechanism string   `default:"kernel" desc:"Default Mechanism to use, supported values kernel,memif" split_words:"true"`
 
-	NetworkServices []*NetworkServiceConfig `default:"" desc:"A list of Network Service Requests with format [{mechanism}]?:${nsName}[@domainName]?/${interfaceName/memIfSocketName}?${label1}=${value1}&${label2}=${value2}" split_words:"true"`
+	NetworkServices []NetworkServiceConfig `default:"" desc:"A list of Network Service Requests with format [{mechanism}]?:${nsName}[@domainName]?/${interfaceName/memIfSocketName}?${label1}=${value1}&${label2}=${value2}" split_words:"true"`
 }
 
 // IsValid - check if configuration is valid

